test/functional/lua/fs_spec.lua

```diff
@@ -308,26 +310,133 @@ describe('vim.fs', function()
       )
     end)
 
-    it('works with UNC paths', function()
-      eq('//foo', vim.fs.normalize('//foo')) -- UNC path
-      eq('//foo/bar', vim.fs.normalize('//foo//bar////')) -- UNC path
-      eq('/foo', vim.fs.normalize('///foo')) -- Not a UNC path
-      eq('/', vim.fs.normalize('//')) -- Not a UNC path
-      eq('/', vim.fs.normalize('///')) -- Not a UNC path
-      eq('/foo/bar', vim.fs.normalize('/foo//bar////')) -- Not a UNC path
+    -- Opts required for testing posix paths and win paths
+    local posix_opts = is_os('win') and { win = false } or {}
+    local win_opts = is_os('win') and {} or { win = true }
+
+    it('preserves leading double slashes in POSIX paths', function()
+      eq('//foo', vim.fs.normalize('//foo', posix_opts))
+      eq('//foo/bar', vim.fs.normalize('//foo//bar////', posix_opts))
+      eq('/foo', vim.fs.normalize('///foo', posix_opts))
+      eq('//', vim.fs.normalize('//', posix_opts))
+      eq('/', vim.fs.normalize('///', posix_opts))
+      eq('/foo/bar', vim.fs.normalize('/foo//bar////', posix_opts))
     end)
 
-    if is_os('win') then
-      it('Last slash is not truncated from root drive', function()
-        eq('C:/', vim.fs.normalize('C:/'))
+    it('allows backslashes on unix-based os', function()
+      eq('/home/user/hello\\world', vim.fs.normalize('/home/user/hello\\world', posix_opts))
+    end)
+
+    it('preserves / after drive letters', function()
+      eq('C:/', vim.fs.normalize([[C:\]], win_opts))
+    end)
+
+    it('works with UNC and DOS device paths', function()
```

Comment:
all of the input/outputs in this test batch look the same (except `\` is coverted to `/`). maybe the inputs should be a bit more complex, e.g. add some double slashes, trailing slashes, etc.